Makefile

```diff
@@ -278,6 +278,14 @@ build: $(BUILD)/fmt ## ensure all packages build
 	go build ./...
 	$Q # caution: some errors are reported on stdout for some reason
 	go test -exec true ./... >/dev/null
+	$Q # compare the current go version with what is in go.mod, and get `{gomod_version}.1` or similar and ensure it builds.
+	$Q GOVERSION="$(shell go env GOVERSION | cut -d. -f1-2)" \
+		MODVERSION="go$(shell grep '^go 1' go.mod | cut -d' ' -f2)"; \
+		if [ "$$GOVERSION" != "$$MODVERSION" ]; then \
```

Comment:
so this was the source of some rather disturbing TILs:

```bash
❯ bash -ec '
[[[ "$a" = "foo"]] && echo first
echo second
'
bash: line 2: [[[: command not found
second

❯ echo $?
0
```

Apparently syntax errors are (sometimes) not script failures, even with `set -e`.

And `sh` does not support `[[`, so it was _printing an error_ and continuing without erroring.  CI uses `sh`, locally we generally get bash or zsh.

Madness.  Absolute madness.  This is going to haunt me for a while, unless I can find a way to prevent it (without shellcheck).